packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -723,6 +723,18 @@ class IOSDevice extends Device {
       return LaunchResult.failed();
     } finally {
       startAppStatus.stop();
+
+      if ((isCoreDevice || forceXcodeDebugWorkflow) && debuggingOptions.debuggingEnabled && package is BuildableIOSApp) {
```

Comment:
for my own knowledge, what determines the value of `debuggingOptions.debuggingEnabled`? Is it just the build mode, are there more complicated rules?

Reply:
Also, in what situations would we reach this far and `package is! BuildableIOSApp`?

Reply:
For your first question: `debuggingOptions.debuggingEnabled` is only determined by the build mode. Release mode would be `false` and debug and profile mode would be true.

https://github.com/flutter/flutter/blob/cb4b4d4ac93981d9f22aa17ce5e10b22346b4700/packages/flutter_tools/lib/src/commands/run.dart#L236-L309

Reply:
For your second question: it will reach this far always regardless of if it's `BuildableIOSApp` or `PrebuiltIOSApp`. We check if it's a `BuildableIOSApp` because we only alter the build settings in the first place if it's a `BuildableIOSApp`
https://github.com/flutter/flutter/blob/cb4b4d4ac93981d9f22aa17ce5e10b22346b4700/packages/flutter_tools/lib/src/ios/devices.dart#L838-L850

Reply:
ahh, thanks.